docgen/src/guides/routing.md

```diff
@@ -0,0 +1,258 @@
+        if (routeStateString === undefined) return {};
+        const routeStateValues = routeStateString.match(/^q\/(.*?)\/brands\/(.*?)\/p\/(.*?)$/);
+        return {
+          q: decodeURIComponent(routeStateValues[1]),
+          brands: decodeURIComponent(routeStateValues[2]),
+          p: routeStateValues[3]
+        }
+      },
+    }),
+    stateMapping: {
+      stateToRoute(uiState) {
+        return {
+          q: uiState.query || '',
+          brands: uiState.refinementList && uiState.refinementList.brand.join('~') || 'all',
+          p: uiState.page || 1
+        };
+      },
+      routeToState(routeState) {
+        if (routeState.brands === 'all') routeState.brands = undefined;
+
+        return {
+          query: routeState.q,
+          refinementList: {brand: routeState.brands && routeState.brands.split('~')},
+          page: routeState.p
+        };
+      }
+    }
+  }
+});
+```
+
+As you can see, we are now using the [instantsearch.routers.history](routing.html#instantsearchroutershistory-api) so that we can explicitly set options on the default router mechanism used in the previous example. What we see also is that both the `router` and `stateMapping` options can be used together as a way to easily map `uiState` to `routeState` and vice versa.
+
+Using that we can configure:
+- `windowTitle`: This method can be used to map the object (`routeState`) returned from `stateToRoute` to your window title
+- `createURL`: This method is called everytime we need to create a url. When we want to synchronize the `routeState` to the browser url bar, when we want to render `<a href>` tags in the `menu` widget or when you call `createURL` in one of your connectors's rendering method
+- `parseURL`: This method is called everytime the user loads, reloads the page or click on back/next buttons of the browser
+
+### About SEO
+
+For your search results to be part of search engines results, you will have to selectively choose them. Trying to have all of your search results inside search engines could be considered as spam by them.
+
+To do that, you can create a [`robots.txt`](http://www.robotstxt.org/) and host it at `https://website.com/robots.txt`.
+
+Here's an example one based on the url scheme we created.
+
+```txt
+User-agent: *
+Allow: /search/q/phones/brands/Samsung/p1
+Allow: /search/q/phones/brands/Apple/p1
+Disallow: /search/
+Allow: *
+```
+
+Now that you know how to create such advanced url synchronization mechanism, the only next step would be to create your own router. But let's first have a look at the full `uiState` reference.
+
+## `instantsearch.routers.history` API
+
+InstantSearch.js provides a default router under `instantsearch.routers.history`. You can use it when you want to go futher than just aliasing querystring parameters in the url. For example if you want to generate urls like `https://website.com/search/q/phone/brands/Sony~Samsung/p/1`.
```

Comment:
`querystring parameters` → `query string parameters`

(also change references below if you decide to change it here)

We should probably add a link to the [definition of a Query String](https://en.wikipedia.org/wiki/Query_string) since this is the first time we talk about it in this guide.